src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Semantics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Summarizes whether a conversion is allowed, and if so, which kind of conversion (and in 
+    /// some cases, the associated symbol).
+    /// </summary>
+    public struct ConversionInfo : IEquatable<ConversionInfo>
+    {
+        private enum ConversionKind
+        {
+            Exists =            0b0000000000000001,
+            IsIdentity =        0b0000000000000010,
+            IsWidening =        0b0000000000000100,
+            IsNarrowing =       0b0000000000001000,
+            IsNumeric =         0b0000000000010000,
+            IsNullable =        0b0000000000100000,
+            IsReference =       0b0000000001000000,
+            IsDefaultLiteral =  0b0000000010000000,
+            IsUserDefined =     0b0000000100000000,
+        }
+
+        private readonly ConversionKind _kind;
+
+        internal ConversionInfo(
+            bool exists, bool isIdentity, bool isWidening, bool isNarrowing, bool isNumeric,
+            bool isNullable, bool isReference, bool isDefaultLiteral, bool isUserDefined,
+            IMethodSymbol methodSymbol)
+        {
+            _kind =
+                (exists ? ConversionKind.Exists : 0) |
+                (isIdentity ? ConversionKind.IsIdentity : 0) |
+                (isWidening ? ConversionKind.IsWidening : 0) |
+                (isNarrowing ? ConversionKind.IsNarrowing : 0) |
+                (isNumeric ? ConversionKind.IsNumeric : 0) |
+                (isNullable ? ConversionKind.IsNullable : 0) |
+                (isReference ? ConversionKind.IsReference : 0) |
+                (isDefaultLiteral ? ConversionKind.IsDefaultLiteral : 0) |
+                (isUserDefined ? ConversionKind.IsUserDefined : 0);
+
+            MethodSymbol = methodSymbol;
+        }
+
+        public override bool Equals(object obj)
+            => obj is ConversionInfo && Equals((ConversionInfo)obj);
+
+        public bool Equals(ConversionInfo other)
+            => _kind == other._kind &&
+               EqualityComparer<IMethodSymbol>.Default.Equals(this.MethodSymbol, other.MethodSymbol);
```

Comment:
MethodSymbols are never value types, or they are already boxed since we are dealing with interface instances. I think object.Equals would be preferred because it is shorter and people generally intuitively understand what it does.